openapi.yaml

```diff
@@ -0,0 +1,2411 @@
+
+    # Common data type
+    ScheduleInterval:
+      oneOf:
+        - $ref: '#/components/schemas/TimeDelta'
+        - $ref: '#/components/schemas/RelativeDelta'
+        - $ref: '#/components/schemas/CronExpression'
+      discriminator:
+        propertyName: objectType
+
+    TimeDelta:
+      type: object
+      properties:
+        objectType: {type: string}
+        days: {type: integer}
+        seconds: {type: integer}
+        microsecond: {type: integer}
+
+    RelativeDelta:
+      # TODO: Why we need these fields?
+      type: object
+      properties:
+        objectType: {type: string}
+        years: {type: integer}
+        months: {type: integer}
+        days: {type: integer}
+        leapdays: {type: integer}
+        hours: {type: integer}
+        minutes: {type: integer}
+        seconds: {type: integer}
+        microseconds: {type: integer}
+        year: {type: integer}
+        month: {type: integer}
+        day: {type: integer}
+        hour: {type: integer}
+        minute: {type: integer}
+        second: {type: integer}
+        microsecond: {type: integer}
+
+    CronExpression:
+      type: object
+      properties:
+        objectType: {type: string}
+        value: {type: string}
+
+    Timezone:
+      type: string
+
+    Tag:
+      # Object to maintain extensibility
+      type: object
+      properties:
+        name:
+          type: string
+
+    Color:
+      type: string
+      pattern: ^#[a-fA-F0-9]{3,6}$
+
+    ClassReference:
+      type: object
+      properties:
+        module_path:
+          type: string
+          readOnly: true
+        class_name:
+          type: string
+          readOnly: true
+
+    # Generic
+    Error:
+      description: >
+        [RFC7807](https://tools.ietf.org/html/rfc7807) compliant response
+      type: object
+      properties:
+        type:
+          type: string
+          description: >
+            A URI reference [RFC3986] that identifies the problem type. This specification
+            encourages that, when dereferenced, it provide human-readable documentation for
+            the problem type.
+        title:
+          type: string
+          description: A short, human-readable summary of the problem  type.
+        status:
+          type: number
+          description: The HTTP status code generated by the API server for this occurrence of the problem.
+        details:
+          type: string
+          description: A human-readable explanation specific to this occurrence of the problem.
+        instance:
+          type: string
+          description: >
+            A URI reference that identifies the specific occurrence of the problem. It may or may
+            not yield further information if dereferenced.
+      required:
+        - type
+        - title
+        - status
+
+    CollectionInfo:
+      type: object
+      properties:
+        total_entries:
+          type: integer
```

Comment:
A generic point: This means we have to do a (possibly expensive?) count operation for every page.

Would it be a better idea instead to have a flag saying "more pages", we then only need to select limit+1 rows to find out if there are more pages than the API caller asked for.

i.e. something like this in pseudo-python

```python
  queryset = session.query(X).offset(req.offset).limit(req.limit + 1)

  ret = { 'more_pages': False, x: [] }
  for n, row in enumerate(queryset):
    if n == req.limit:
      ret['more_pages'] = True
      break
    ret['x'].append(row.to_api_representation())

```

Reply:
This will make it harder to use the API, e.g. it will make it difficult to create pagination, or it will make it difficult to download many pages at the same time. 
![Screenshot 2020-05-26 at 00 59 15](https://user-images.githubusercontent.com/12058428/82847043-2c3eb900-9eec-11ea-9659-7d609ce01acc.png)
![Screenshot 2020-05-26 at 01 06 55](https://user-images.githubusercontent.com/12058428/82847222-36ad8280-9eed-11ea-86e5-bb2d4cb49b92.png)

Infinite pagination is useful when building an API for a site that is very often used and we need to optimize database usage very drastically or when we expect users to never go to the last page, e.g. a news site.  However, Airflow is a system used inside organizations, and therefore has relatively few users. Counting all objects should not be a problem. if this is a problem, I think it is worth to think about how to optimize it.